### ISSUE TITLE
xen-hypervisor: provide a debug=y build of Xen

### ISSUE THREAD

**Issue by royger** [QubesOS/qubes-issues#9676]
### The problem you're addressing (if any)

When attempting to debug a Xen issue it's common practice to ask the user to switch to a debug build of Xen (a Xen hypervisor built with `CONFIG_DEBUG=y`).  Building such debug enabled hypervisor can be difficult for users, so it would be best to provide one in QubesOS.

### The solution you'd like

Users should be able to switch to a debug build of Xen without having to resort to build it themselves.  FreeBSD for example builds the hypervisor twice to create a debug=n and debug=y versions of Xen, and installs the later as `xen-debug`:

https://cgit.freebsd.org/ports/tree/emulators/xen-kernel/Makefile

### The value to a user, and who that user might be

Make it easier to switch to debug build of Xen, and as a result provide more helpful information for debugging hypervisor issues.

**Comment by marmarek** [comment]
Almost duplicate of https://github.com/QubesOS/qubes-issues/issues/5989

Anyway, there is a PR implementing this already: https://github.com/QubesOS/qubes-vmm-xen/pull/170. But it's blocked on bootloader ordering: https://github.com/QubesOS/qubes-vmm-xen/pull/170#issuecomment-1839749393
